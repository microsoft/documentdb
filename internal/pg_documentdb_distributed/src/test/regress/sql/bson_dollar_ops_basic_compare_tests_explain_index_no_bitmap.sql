
set search_path to helio_core,helio_api,helio_api_catalog,pg_catalog;
SET citus.next_shard_id TO 510000;
SET helio_api.next_collection_id TO 5100;
SET helio_api.next_collection_index_id TO 5100;

SELECT helio_api.drop_collection('db', 'querydollartest') IS NOT NULL;
SELECT helio_api.create_collection('db', 'querydollartest');

SET client_min_messages=WARNING;

-- Create a wildcard index by using CREATE INDEX command instead of
-- using helio_api.create_indexes. This is because, we will use
-- that index to test whether we can use the index via query operators
-- other than "@@".
SELECT helio_api_internal.create_indexes_non_concurrently('db', helio_distributed_test_helpers.generate_create_index_arg('querydollartest', 'index_2', '{"$**": 1}'), TRUE);
SELECT helio_distributed_test_helpers.drop_primary_key('db', 'querydollartest');


BEGIN;
-- avoid sequential scan (likely to be preferred on small tables)
set local enable_seqscan to off;
set local helio_api.forceUseIndexIfAvailable to on;
set local enable_bitmapscan to off;
set local helio_api.forceRumIndexScantoBitmapHeapScan TO OFF;

\i sql/bson_dollar_ops_basic_compare_tests_explain_core.sql
ROLLBACK;
