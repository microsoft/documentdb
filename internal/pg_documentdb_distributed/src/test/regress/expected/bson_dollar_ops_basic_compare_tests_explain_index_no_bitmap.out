NOTICE:  identifier "pg_regress/bson_dollar_ops_basic_compare_tests_explain_index_no_bitmap" will be truncated to "pg_regress/bson_dollar_ops_basic_compare_tests_explain_index_no"
set search_path to helio_core,helio_api,helio_api_catalog,pg_catalog;
SET citus.next_shard_id TO 510000;
SET helio_api.next_collection_id TO 5100;
SET helio_api.next_collection_index_id TO 5100;
SELECT helio_api.drop_collection('db', 'querydollartest') IS NOT NULL;
 ?column? 
---------------------------------------------------------------------
 t
(1 row)

SELECT helio_api.create_collection('db', 'querydollartest');
NOTICE:  creating collection
 create_collection 
---------------------------------------------------------------------
 t
(1 row)

SET client_min_messages=WARNING;
-- Create a wildcard index by using CREATE INDEX command instead of
-- using helio_api.create_indexes. This is because, we will use
-- that index to test whether we can use the index via query operators
-- other than "@@".
SELECT helio_api_internal.create_indexes_non_concurrently('db', helio_distributed_test_helpers.generate_create_index_arg('querydollartest', 'index_2', '{"$**": 1}'), TRUE);
                                                                                                   create_indexes_non_concurrently                                                                                                    
---------------------------------------------------------------------
 { "raw" : { "defaultShard" : { "numIndexesBefore" : { "$numberInt" : "1" }, "numIndexesAfter" : { "$numberInt" : "2" }, "createdCollectionAutomatically" : false, "ok" : { "$numberInt" : "1" } } }, "ok" : { "$numberInt" : "1" } }
(1 row)

SELECT helio_distributed_test_helpers.drop_primary_key('db', 'querydollartest');
 drop_primary_key 
---------------------------------------------------------------------
 
(1 row)

BEGIN;
-- avoid sequential scan (likely to be preferred on small tables)
set local enable_seqscan to off;
set local helio_api.forceUseIndexIfAvailable to on;
set local enable_bitmapscan to off;
set local helio_api.forceRumIndexScantoBitmapHeapScan TO OFF;
\i sql/bson_dollar_ops_basic_compare_tests_explain_core.sql
/* validate explain */
EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @= '{ "a.b": 1 }';
                                                        QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@=) '{ "a.b" : { "$numberInt" : "1" } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @!= '{ "a.b": 1 }';
                                                         QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@!=) '{ "a.b" : { "$numberInt" : "1" } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @> '{ "a.b": 1 }';
                                                        QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@>) '{ "a.b" : { "$numberInt" : "1" } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @>= '{ "a.b": 1 }';
                                                         QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@>=) '{ "a.b" : { "$numberInt" : "1" } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @< '{ "a.b": 1 }';
                                                        QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@<) '{ "a.b" : { "$numberInt" : "1" } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @<= '{ "a.b": 1 }';
                                                         QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@<=) '{ "a.b" : { "$numberInt" : "1" } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @*= '{ "a.b": [ 1, 2, true ]}'::bson;
                                                                          QUERY PLAN                                                                          
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@*=) '{ "a.b" : [ { "$numberInt" : "1" }, { "$numberInt" : "2" }, true ] }'::helio_core.bson)
(7 rows)

/* validate explain for the individual items as well */
EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @> '{ "a": { "b": 1 } }';
                                                            QUERY PLAN                                                             
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@>) '{ "a" : { "b" : { "$numberInt" : "1" } } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @> '{ "a": { "b": [ true, false ] } }';
                                                         QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@>) '{ "a" : { "b" : [ true, false ] } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @> '{ "a": [ { "b": [ 2, 3, 4 ] } ] }';
                                                                                        QUERY PLAN                                                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@>) '{ "a" : [ { "b" : [ { "$numberInt" : "2" }, { "$numberInt" : "3" }, { "$numberInt" : "4" } ] } ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @>= '{ "a": { "b": 1 } }';
                                                             QUERY PLAN                                                             
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@>=) '{ "a" : { "b" : { "$numberInt" : "1" } } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @>= '{ "a": { "b": [ true, false ] } }';
                                                         QUERY PLAN                                                          
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@>=) '{ "a" : { "b" : [ true, false ] } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @>= '{ "a": [ { "b": [ 2, 3, 4 ] } ] }';
                                                                                         QUERY PLAN                                                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@>=) '{ "a" : [ { "b" : [ { "$numberInt" : "2" }, { "$numberInt" : "3" }, { "$numberInt" : "4" } ] } ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @< '{ "a": { "b": 1 } }';
                                                            QUERY PLAN                                                             
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@<) '{ "a" : { "b" : { "$numberInt" : "1" } } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @< '{ "a": { "b": [ true, false ] } }';
                                                         QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@<) '{ "a" : { "b" : [ true, false ] } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @< '{ "a": [ { "b": [ 2, 3, 4 ] } ] }';
                                                                                        QUERY PLAN                                                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@<) '{ "a" : [ { "b" : [ { "$numberInt" : "2" }, { "$numberInt" : "3" }, { "$numberInt" : "4" } ] } ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @<= '{ "a": { "b": 1 } }';
                                                             QUERY PLAN                                                             
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@<=) '{ "a" : { "b" : { "$numberInt" : "1" } } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @<= '{ "a": { "b": [ true, false ] } }';
                                                         QUERY PLAN                                                          
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@<=) '{ "a" : { "b" : [ true, false ] } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @<= '{ "a": [ { "b": [ 2, 3, 4 ] } ] }';
                                                                                         QUERY PLAN                                                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@<=) '{ "a" : [ { "b" : [ { "$numberInt" : "2" }, { "$numberInt" : "3" }, { "$numberInt" : "4" } ] } ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @= '{ "a": { "b": 1 } }';
                                                            QUERY PLAN                                                             
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@=) '{ "a" : { "b" : { "$numberInt" : "1" } } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @= '{ "a": { "b": [ true, false ] } }';
                                                         QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@=) '{ "a" : { "b" : [ true, false ] } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @= '{ "a": [ { "b": [ 2, 3, 4 ] } ] }';
                                                                                        QUERY PLAN                                                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@=) '{ "a" : [ { "b" : [ { "$numberInt" : "2" }, { "$numberInt" : "3" }, { "$numberInt" : "4" } ] } ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @!= '{ "a": { "b": 1 } }';
                                                             QUERY PLAN                                                             
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@!=) '{ "a" : { "b" : { "$numberInt" : "1" } } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @!= '{ "a": { "b": [ true, false ] } }';
                                                         QUERY PLAN                                                          
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@!=) '{ "a" : { "b" : [ true, false ] } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @!= '{ "a": [ { "b": [ 2, 3, 4 ] } ] }';
                                                                                         QUERY PLAN                                                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@!=) '{ "a" : [ { "b" : [ { "$numberInt" : "2" }, { "$numberInt" : "3" }, { "$numberInt" : "4" } ] } ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @*= '{ "a.b": [ { "b": [ 2, 3, 4 ] }, 2, true ]}';
                                                                                                         QUERY PLAN                                                                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@*=) '{ "a.b" : [ { "b" : [ { "$numberInt" : "2" }, { "$numberInt" : "3" }, { "$numberInt" : "4" } ] }, { "$numberInt" : "2" }, true ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @!*= '{ "a.b": [ { "b": [ 2, 3, 4 ] }, 2, true ]}';
                                                                                                         QUERY PLAN                                                                                                          
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@!*=) '{ "a.b" : [ { "b" : [ { "$numberInt" : "2" }, { "$numberInt" : "3" }, { "$numberInt" : "4" } ] }, { "$numberInt" : "2" }, true ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @? '{ "a.b": 1 }';
                                                        QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@?) '{ "a.b" : { "$numberInt" : "1" } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @? '{ "a.b": 0 }';
                                                        QUERY PLAN                                                         
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@?) '{ "a.b" : { "$numberInt" : "0" } }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @&= '{ "a.b": [1] }';
                                                           QUERY PLAN                                                           
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@&=) '{ "a.b" : [ { "$numberInt" : "1" } ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @&= '{ "a.b": [0, 1] }';
                                                                       QUERY PLAN                                                                       
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@&=) '{ "a.b" : [ { "$numberInt" : "0" }, { "$numberInt" : "1" } ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @&= '{ "a.b": [[0]] }';
                                                             QUERY PLAN                                                             
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@&=) '{ "a.b" : [ [ { "$numberInt" : "0" } ] ] }'::helio_core.bson)
(7 rows)

EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db', 'querydollartest') WHERE document @&= '{ "a.b": [] }';
                                                QUERY PLAN                                                
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Index Scan using index_2 on documents_5100_510006 collection
               Index Cond: (document OPERATOR(helio_api_catalog.@&=) '{ "a.b" : [  ] }'::helio_core.bson)
(7 rows)

ROLLBACK;
